File: C Programming/problems on recursion/problem8.c
/*
8. Write a recursive program which accept number from user and return its factorial.  
	
	Input :  5 
	
	Output : 120 
		
	Author :Milind Nalawade
	
	Date : 25/08/2021
	
	Language : C programming
	
*/

#include<stdio.h>

//Logic
int factor=1;
int Factor(int no)
{
	if(no==1)
	{
		return factor;
	}
	else
	{
		factor*=no;
		no--;
		Factor(no);
	}
}	

//Entry point function
int main()
{
	int no=0,ret=0;
	printf("\n Input : ");
	scanf("%d",&no);
	ret=Factor(no);
	printf("\n Output : %d \n",ret);
	return 0;
}	

File: Python Programming/problems on recursion/problem8.py
'''
8. Write a recursive program which accept number from user and return its factorial.  
	
	Input :  5 
	
	Output : 120 
	
	Author :Milind Nalawade
	
	Date : 25/08/2021
	
	Language : Python 3.9.0

'''
factor=1
#====
#Logic
#====
def Output(no):
	global factor 
	if no==1:
		return factor
	else:
		factor*=no
		Output(no-1)
		
#=============
#Entry point function
#=============
def main():
	global factor
	no=int(input("\n Input : "))
	Output(no)
	print("\n Output : {} \n".format(factor))
	
#=====
#starter
#=====
if __name__=="__main__":
	main()